crates/intrinsic-test/src/common/argument.rs

```diff
@@ -106,11 +111,13 @@ where
         loads: u32,
     ) -> std::io::Result<()> {
         for arg in self.iter().filter(|&arg| !arg.has_constraint()) {
```

Comment:
I note that arguments of the same type seem to have identical `_vals` arrays. Not required for this but as a future optimisation we could just emit one array for each unique argument type